### PR TITLE
chore(tls): add tls config test for ca_dir

### PR DESCRIPTION
**Pull request by kostasrim**
Partially addresses #2081 by adding a test case to cover the use case

* add test for `ca_dir` 

I will update our docs as well.

### CODE REVIEW

File: tests/dragonfly/conftest.py [hunk 1]
```diff
@@ -286,6 +286,17 @@ def with_ca_tls_server_args(with_tls_server_args, with_tls_ca_cert_args):
     return args
 
 
+@pytest.fixture(scope="session")
+def with_ca_dir_tls_server_args(with_tls_server_args, with_tls_ca_cert_args):
+    args = deepcopy(with_tls_server_args)
+    ca_cert = with_tls_ca_cert_args["ca_cert"]
+    ca_dir = os.path.dirname(ca_cert)
+    command = f"c_rehash {ca_dir}"
```
Comment (romange) [inline tests/dragonfly/conftest.py:294]: maybe worth adding a comment on why it's needed.

**Comment by romange** [comment]
Also, both the issue and the PR description do not talk about the root cause. it's ok for the issue because the issuer does not have to do root cause analysis, but it could be nice to expand in the commit description of what was wrong and what's addressed.

**Comment by kostasrim** [comment]
> Also, both the issue and the PR description do not talk about the root cause. it's ok for the issue because the issuer does not have to do root cause analysis, but it could be nice to expand in the commit description of what was wrong and what's addressed.

I will add this when we commit it: 
```
    This PR introduces a test case for TLS with `ca_dir`. First, we
    did not have any tests for this case. Second, using `ca_dir` requires
    to call `c_rehash` on the directory before it is loaded by DF. We
    did not have this use case anywhere and therefore we thought there was
    a bug when we used `ca_dir` only to find out that we need to call
    `c_rehash` on the directory before we load the certificates. Now,
    both the test and the use case are properly documented
    
    * add missing test for ca_dir
    * use rehash to properly show how to load ca directories instead of
      files
```

**Comment by romange** [comment]
Ah, got ya. The test is for demonstration/coverage purposes.

**Review by romange — APPROVED** [review]

**Pull request merged**